openshift/image-inspector-template.yaml

```diff
@@ -0,0 +1,99 @@
+apiVersion: "v1"
+kind: "Template"
+metadata:
+  name: "image-inspector-scan"
+  annotations:
+    description:
+      Template for scanning a container image. Requires privileged permissions.
+    tags: infrastructure
+objects:
+- apiVersion: "v1"
+  kind: "Pod"
+  metadata:
+    name: "${NAME}"
+    labels:
+      name: "${NAME}"
+    annotations:
+      openshift.io/scc: "privileged"
+      openshift.io/image-scan-target: "${TARGET_IMAGE}"
+  spec:
+    containers:
+    - name: "image-inspector"
+      image: "${IMAGE_REGISTRY}/${IMAGE_NAME}:${IMAGE_TAG}"
+      imagePullPolicy: "Always"
+      command:
+      - "/usr/bin/image-inspector"
+      - "--chroot"
+      - "--username=${SERVICE_ACCOUNT}"
+      - "--password-file=${TOKEN_FILE}"
+      - "--image=${TARGET_IMAGE}"
+      - "--scan-type=openscap"
+      - "--cve-url=${CVE_URL}"
+      - "--openscap-html-report"
+      - "--serve=:8080"
+      ports:
+      - containerPort: 8080
+        protocol: "TCP"
+      volumeMounts:
+      - name: "docker-socket"
+        mountPath: "${DOCKER_SOCKET}"
+      securityContext:
+        privileged: true
+      readinessProbe:
+        httpGet:
+          path: "/healthz"
+          port: 8080
+        initialDelaySeconds: 15
+        periodSeconds: 5
+    nodeSelector: "${{NODE_SELECTOR}}"
+    automountServiceAccountToken: true
+    serviceAccountName: "${SERVICE_ACCOUNT}"
```

Comment:
@pweil- do you know if it's possible to mount the dockercfg instead of the token? I remember we worked on this a lot with @enoodle. IIRC that was because the dockercfg file could contain multiple docker login credentials.

Reply:
Mounting a file is possible.  The complication here is that you need access to potentially private registries and the secrets are held in some other namespace.  Users themselves will be using secrets for this. What is the plan to access their secrets?

Reply:
@pweil- I was referring to the standard `default-dockercfg-*` secret in the `management-infra` namespace. Is there a way to reference and mount that instead of the token? (Name is random, so it's hard to define the mount here in the template, and `automountServiceAccountToken` is mounting the token not the dockercfg).
Although, yes, we could try to support another parameter `DOCKERCFG_SECRET_NAME` in case someone wants to tweak it.

Reply:
You'd need to specify the secret by name to mount it as a volume.  

I guess the issue still exists though.   If the secret in management-infra doesn't contain *every* docker auth setting in the entire cluster there will still be pull errors.  Maybe that's not an issue you're trying to solve here, it just means that some images will be unscannable unless it is done on the node where the image already exists and the pull policy is "if not present"

Reply:
@pweil- so it seems that juggling PVs for each single scan is an overhead. Should we try to move to a worker-model where we define in advance how many worker we have (each with a dedicated PV) and they'll pick up requests and serve responses for a specific image?
It seems some considerable work.

@smarterclayton any suggestion from your side?